juspay/hyperswitch
# feat(router): make core changes in payments flow to support incremental authorization

## Type of Change



## Description
**Incremental authorizations** allows you to increase authorized amount for a payment before capturing it
- This is possible only for card payments and specific networks, with some criteria specific to connector
- Incremental Authorizations can be done multiple times and maximum attempts is again based on connector 
- Total amount capturable would be sum of initial authorized amount and all the incremented additional amounts

1. accept `request_incremental_authorization` flag in payments request
2. send the corresponding value in payments_authorize_data to connecor flows
3. receive `incremental_authorization_allowed` from connector flows
4. store them in payment_intent and send incremental_authorization_allowed back in payments response


### Additional Changes





## Motivation and Context



## How did you test it?
Tested Manually
1. send request_incremental_authorization as true in request, with capture_method automatic. should fail with an error
<img width="1246" alt="image" src="https://github.com/juspay/hyperswitch/assets/56996463/33da6d41-00f1-431c-b554-ef884de3e73c">

2. send request_incremental_authorization as false in request, see incremental_authorization_allowed being false in response for successful payment in Cybersouce
<img width="1246" alt="image" src="https://github.com/juspay/hyperswitch/assets/56996463/627a51f2-3640-4cc6-91fe-34068dda394a">

3. send request_incremental_authorization as true in request, see incremental_authorization_allowed being true in response for successful payment in Cybersouce
<img width="1246" alt="image" src="https://github.com/juspay/hyperswitch/assets/56996463/2c0f7ca7-f03e-437c-ae15-25a6df2e30e5">

## Checklist
